pkg/caveats/eval_test.go

```diff
@@ -167,44 +169,26 @@ func TestEvaluateCaveat(t *testing.T) {
 		},
 		{
 			"map evaluation",
-			mustEnvForVariables(map[string]VariableType{
-				"m":   MapType(IntType, BooleanType),
-				"idx": IntType,
-			}),
-			"m[idx]",
-			map[string]any{
-				"m": map[int]bool{
-					1: true,
-				},
-				"idx": 1,
-			},
-			"",
-			true,
-			"",
-			noMissingVars,
-		},
-		{
-			"map evaluation, wrong map kind",
-			mustEnvForVariables(map[string]VariableType{
-				"m":   MapType(IntType, BooleanType),
-				"idx": IntType,
+			mustEnvForVariables(map[string]types.VariableType{
+				"m":   types.MapType(types.BooleanType),
+				"idx": types.StringType,
 			}),
 			"m[idx]",
 			map[string]any{
 				"m": map[string]bool{
 					"1": true,
 				},
-				"idx": 1,
+				"idx": "1",
 			},
-			"no such key: 1",
```

Comment:
huh, shouldn't this have passed before the changes in this PR?

Reply:
No, since the map's index was an *int*, and now its a string

Reply:
ah, duh 🤦🏻